app/controllers/concerns/hyku_addons/users_controller_behavior.rb

```diff
@@ -12,7 +12,15 @@ def index
 
     def show
       @user = User.find_by(email: params[:email], display_profile: true)
-      render json: { status: 403, code: 'forbidden', message: t("errors.users_forbidden") } if @user.blank?
+      render json: { status: 403, code: 'forbidden', message: t("errors.users_forbidden") } and return if @user.blank?
+
+      query_string = "(creator_tesim:\"*#{@user.email}*\") AND visibility_ssi:open"
+      @user_document_id_list = ActiveFedora::SolrService.get(query_string, rowy: 1_000_000)['response']['docs'].pluck('id')
+      @user_works = Hyrax::PresenterFactory.build_for(ids: @user_document_id_list,
+                                                      presenter_class: Hyku::WorkShowPresenter,
+                                                      presenter_args: current_ability)
+      collection_search_builder = Hyrax::CollectionSearchBuilder.new(self).with_access(:read).rows(1_000_000)
+      @collection_docs = repository.search(collection_search_builder).documents
```

Comment:
Are you using all three `@user_document_id_list`, `@user_works` and `@collection_docs` in the view or in other methods inside of the request?

The entire block of code could be moved to a seperate protected method to reduce the noise and if only a single instance variable is being used to pass data to the view, then the others can be made normal method variables, which means the garbage collection could potentially pick them up faster and reduce the app memory usage.

Reply:
I am not super sure :sweat_smile: I found a similar block of code in the engine file and used that. I will add it to the list of things to look at and refactor :smiley:

Reply:
You should try not to just copy and paste. Writing the code out by hand might seem more long winded, but you get several benefits from doing it. You'll remember the code many times quicker than copying/pasting it and you'll notice what is and is not required for your specific context.

Reply:
`@user_works` and `@collection_docs` are both used in the view. I don't think its a good idea to assign instance variables in protected methods is it?